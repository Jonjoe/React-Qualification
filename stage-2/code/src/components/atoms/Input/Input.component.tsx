import * as React from "react";

function Input(): JSX.Element {
  return <input />;
}

export default Input as React.FC;
