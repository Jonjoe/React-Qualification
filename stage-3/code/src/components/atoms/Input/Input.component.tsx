import React from "react";

interface InputProps {}

function Input(props: InputProps): JSX.Element {
  return <input />;
}

export default Input as React.FC<InputProps>;
