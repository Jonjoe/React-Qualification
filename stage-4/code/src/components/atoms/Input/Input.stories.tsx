import React from "react";

import { storiesOf } from "@storybook/react";

import Input from "./Input.component";

storiesOf("ColorButton", module)
    .add("default",
         () => <Input onChange={() => null} value="hello"/>
    )
