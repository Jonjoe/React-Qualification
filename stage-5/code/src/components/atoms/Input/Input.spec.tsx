import React from "react";
import renderer from "react-test-renderer";

import Input from "./Input.component";

test("Link changes the class when hovered", () => {
  const component = renderer.create(
    <Input onChange={() => null} value="Example Value" />
  );

  const tree = component.toJSON();
  
  expect(tree).toMatchSnapshot();
});
