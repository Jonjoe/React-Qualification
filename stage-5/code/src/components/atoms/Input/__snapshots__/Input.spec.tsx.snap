// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<div
  className="sc-AxjAm jgeyuh"
>
  <input
    onChange={[Function]}
    value="Example Value"
  />
</div>
`;
